src/main/java/persistence/sql/dialect/h2/H2Query.java

```diff
@@ -0,0 +1,53 @@
+package persistence.sql.dialect.h2;
+
+import persistence.sql.Query;
+import persistence.sql.ddl.CreateQueryBuilder;
+import persistence.sql.ddl.DropQueryBuilder;
+import persistence.sql.dml.DeleteQueryBuilder;
+import persistence.sql.dml.InsertQueryBuilder;
+import persistence.sql.dml.SelectQueryBuilder;
+import persistence.sql.entity.EntityData;
+
+public class H2Query<T, K> implements Query<T, K> {
+
+    private static final H2Dialect dialect = new H2Dialect();
```

Comment:
> Query의 역할을 분담한다고 했는데, 저는 쿼리 빌더들을 조합하는 무언가가 있어야 하지 않나..? 해서 파사드 형태로 취급하였는데, 요한님이 생각하신 해당 인터페이스의 분리가 구체적으로 어떻게 구현되는지 궁금합니다!

저도 해준님의 생각과 같이 쿼리 빌더들을 조합하는 무언가가 있어야한다고 생각합니다.
조금 스포하자면 다만 다음 미션인 `Entity 구현`에서 다루어질 부분이기 때문에 지금은 현재 단계에서는 요구사항에 맞춰 간단하게 구현하시는걸 추천드립니다 :)

만들어 주신 파사드 형태의 H2Query 이야기도 해보자면 Dialect의 추상화를 이용한다면
H2Query라는 구현체가 없더라도 H2Dialect을 주입하여 H2 에 적절한 쿼리 빌더들을 만들어낼 수 있을 것이라 생각됩니다!
아래 예시 처럼 구현한다면 굳이 데이터베이스 타입별로 Query의 구현체를 만들어내지 않아도 유연하게 사용할 수 있지 않을까요?
(현재 미션의 요구사항에서 벗어나기 때문에 지금 단계에서 아래와 같이 구현해달라고 요구하는 것은 아닙니다 😄 
오히려 지금은 덜어내는 것이 필요할 것 같아요)
```suggestion
public class QueryImpl {
    private Dialect dialect;
    
    public QueryImpl(Dialect dialect) {
        this.dialect = dialect
    }
```

Reply:
오 피드백 감사드립니다! 슬쩍 바꿔보겠습니다 ㅎㅎ